contrib/depends/toolchain.cmake.in

```diff
@@ -49,11 +49,24 @@ set (CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER) # Find programs on host
 set (CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY) # Find libs in target
 set (CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY) # Find includes in target
 
+function(split_program input_string program arguments)
+  # CMake < 3.19 only understands setting CMAKE_<LANG>_COMPILER using "<program>" "<arguments>".
+  # https://cmake.org/cmake/help/v3.19/variable/CMAKE_LANG_COMPILER.html#variable:CMAKE_%3CLANG%3E_COMPILER
+
+  string(FIND "${input_string}" " " space_index)
+
+  string(SUBSTRING "${input_string}" 0 ${space_index} _program)
+  string(SUBSTRING "${input_string}" ${space_index} -1 _arguments)
```

Comment:
I'm not sure it matters if you include a space at the start of arguments, so `space_index` might be fine here; otherwise, you can adjust the index using `MATH(EXPR args_index "${space_index}+1")`). I'd suggest checking if `FIND` matched, otherwise this substring would have invalid bounds (the `_program` one still works fine, as 0 to -1 is the whole string).